### PR TITLE
Adds Void puppies

### PR DESCRIPTION
**Pull request by Shadowlight213**
Requested by and sprites made by @WJohn 
Adds Void puppies, a rare type of corgi puppy that can be born from corgis along with support for rare children for other mobs.
void
:cl:
rscadd: A wave of dark energy has been detected in the area. Corgi births may be affected.
/:cl:

### CODE REVIEW

**Comment by Zilenan91** [comment]
Do they do anything

**Comment by KazeEspada** [comment]
They stare back.

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 1]
```diff
@@ -518,6 +523,19 @@
...
 
 
+/mob/living/simple_animal/pet/dog/corgi/puppy/void		//Tribute to the corgis born in nullspace
+	name = "\improper void puppy"
+	real_name = "voidy"
+	desc = "A corgi puppy that has been infused with deep space energy."
+	icon_state = "void_puppy"
+	icon_living = "void_puppy"
+	icon_dead = "void_puppy_dead"
+	density = 0
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: puppies are already nondense.

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 2]
```diff
@@ -518,6 +523,19 @@
...
+	nofur = 1
+	pass_flags = PASSMOB
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: puppies already have PASSMOB

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 3]
```diff
@@ -518,6 +523,19 @@
...
+	mob_size = MOB_SIZE_SMALL
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: puppies are already MOB_SIZE_SMALL

File: code/modules/mob/living/simple_animal/friendly/dog.dm (hunk 1 above)
Comment (Shadowlight213) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: they are extra nondense

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 4]
```diff
@@ -518,6 +523,16 @@
...
 
 
+/mob/living/simple_animal/pet/dog/corgi/puppy/void		//Tribute to the corgis born in nullspace
+	name = "\improper void puppy"
+	real_name = "voidy"
+	desc = "A corgi puppy that has been infused with deep space energy."
+	icon_state = "void_puppy"
+	icon_living = "void_puppy"
+	icon_dead = "void_puppy_dead"
+	nofur = 1
```
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: >not space proof

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 5]
```diff
@@ -24,11 +24,13 @@
 	gender = MALE
 	butcher_results = list(/obj/item/weapon/reagent_containers/food/snacks/meat/slab/corgi = 3, /obj/item/stack/sheet/animalhide/corgi = 1)
 	childtype = /mob/living/simple_animal/pet/dog/corgi/puppy
+	rarechildtype = /mob/living/simple_animal/pet/dog/corgi/puppy/void
```
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: Should make it a list so other rare types are possible

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 6]
```diff
@@ -55,6 +55,7 @@
 
 	//Hot simple_animal baby making vars
 	var/childtype = null
+	var/rarechildtype = null	//Special children
```
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: Make childtype a weighted list and have the below bit pick from it instead. Would make it easier to add things down the line

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 7]
```diff
@@ -509,7 +510,10 @@
 			alone = 0
 			continue
 	if(alone && partner && children < 3)
-		new childtype(loc)
+		if(prob(5) && rarechildtype)
+			new rarechildtype(loc)
+		else
+			new childtype(loc)
```
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: And pick from the list here instead of the if

Comment (Shadowlight213) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: Except I don't want an equal chance to have the rare one. Rare one needs to be rare.

File: code/modules/mob/living/simple_animal/simple_animal.dm (hunk 6 above)
Comment (Shadowlight213) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: How do I make a weighted list?

**Comment by WJohn** [comment]
It's a meme. Right now we have a byond bug where when corgis breed, there's a good possibility that the puppies end up in null space. Thus, void puppies.

**Comment by RemieRichards** [comment]
No thats not right at all @WJohn 
It's tangentially related to breeding, but not primarily.

the cause is the same as with Cleanbots.
a call to play a sound effect is swapped with a call to create a mob (Cleanbots, now corgis), by a byond error.

The only unique thing about corgis is they can still breed in nullspace, so when one of each gender end up in nullspace they can breed there.

File: code/modules/mob/living/simple_animal/simple_animal.dm (hunk 6 above)
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: Look at /obj/effect/lootspawner for an example of spawning from a weighted list. On my phone so  it'd be a pain to find/make an example for you

**Comment by Crushtoe** [comment]
Disappointed my Sprite wasn't used.
image

File: code/modules/mob/living/simple_animal/simple_animal.dm (hunk 6 above)
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: Weighted list 1: Pure Byond:
```var/list/L = list(prob(prob_1); element_1, prob(prob_2); element_2)```
this uses regular ```pick()```

Weighted list 2: tgcode:
```var/list/L = list(element_1 = prob_1, element_2 = prob_2)```
this uses ```pickweight()```

File: code/modules/mob/living/simple_animal/friendly/dog.dm [hunk 8]
```diff
@@ -518,6 +522,22 @@
...
 
 
+/mob/living/simple_animal/pet/dog/corgi/puppy/void		//Tribute to the corgis born in nullspace
+	name = "\improper void puppy"
+	real_name = "voidy"
+	desc = "A corgi puppy that has been infused with deep space energy. It's staring back.."
+	icon_state = "void_puppy"
+	icon_living = "void_puppy"
+	icon_dead = "void_puppy_dead"
+	nofur = 1
```
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: Generally we use true/false instead of 1/0

Comment (duncathan) [inline code/modules/mob/living/simple_animal/friendly/dog.dm]: `TRUE` and `FALSE`, that is

**Pull request merged**